server/SchemaLoader.js

```diff
@@ -0,0 +1,127 @@
+"use strict";
+
+const EventEmitter = require("events");
+const { remote } = require("electron");
+const jsonfile = require("jsonfile");
+const mkdirp = require("mkdirp");
+const path = require("path");
+const fs = require("fs");
+
+const ValidationSchema = require("./model/ValidationSchema");
+
+class SchemaLoader extends EventEmitter {
+    constructor() {
+        super();
+
+        this.on("schemas-synced", function() {
+            this.loadLocalSchemas();
+        });
+    }
+
+    syncSchemas() {
+        let repo = remote.getGlobal("options").repo;
+
+        if (!!repo && !!repo.url && !!repo.apikey && !!repo.vendors && repo.vendors.length > 0) {
+            let endpoint = "/api/schemas/" + repo.vendors.join("%2C");
+
+            const req = new XMLHttpRequest();
+            req.open("GET", repo.url + endpoint);
+            req.setRequestHeader("apikey", repo.apikey);
+            req.send();
+
+            req.onreadystatechange = function() {
+                if (req.readyState === 4 && req.status === 200) {
+                    //done loading
+                    this.storeSchemas(JSON.parse(req.responseText));
+                    this.emit("schemas-synced");
+                }
+            }.bind(this);
+        } else {
+            this.emit("schemas-synced");
+        }
+    }
+
+    storeSchemas(schemas) {
+        schemas.forEach((schema) => {
+            let content = JSON.stringify(schema);
+            let schemaPath = path.join(this.getSchemasPath(), this.getNameFromSchema(schema));
+
+            // create path (folders) if they don't exist
+            mkdirp(path.dirname(schemaPath), function(mkdirError) {
+                if (mkdirError) {
+                    return console.log(mkdirError);
```

Comment:
- What is the return value of `console.log` used for?
- Shouldn't we throw an error?

Reply:
1) It isn't used as it's an async function, but the `return` stops executing the rest of the function.
2) We can. Will directly use a catch through, as I don't want to prevent other schemas from being stored.

Reply:
Re 1: in that case it's clearer to write:
```js
console.log(mkdirError);
return;
```
:)